From 56c68a50e7970e971136af952c79360c31c72efd Mon Sep 17 00:00:00 2001
From: Sandy Huang <hjc@rock-chips.com>
Date: Mon, 11 Dec 2017 11:52:40 +0800
Subject: [PATCH] drm/rockchip: vop: fix vop_area->area_id define error

1.
before this commit:
	win3-0: DISABLED
	win3-0: DISABLED
	win3-1: DISABLED
	win3-2: DISABLED
after this commit:
	win3-0: DISABLED
	win3-1: DISABLED
	win3-2: DISABLED
	win3-3: DISABLED
2. fix compile error:
    vop_crtc_load_lut() error: uninitialized symbol 'lut_idx'.

Change-Id: I6cd2de972181438004a68d38ffeca939772ddb29
Signed-off-by: Sandy Huang <hjc@rock-chips.com>
---
 drivers/gpu/drm/rockchip/rockchip_drm_vop.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/rockchip/rockchip_drm_vop.c b/drivers/gpu/drm/rockchip/rockchip_drm_vop.c
index f7541a53ccba..a274882fa78d 100644
--- a/drivers/gpu/drm/rockchip/rockchip_drm_vop.c
+++ b/drivers/gpu/drm/rockchip/rockchip_drm_vop.c
@@ -1113,7 +1113,7 @@ static void vop_line_flag_irq_disable(struct vop *vop)
 static void vop_crtc_load_lut(struct drm_crtc *crtc)
 {
 	struct vop *vop = to_vop(crtc);
-	int i, dle, lut_idx;
+	int i, dle, lut_idx = 0;
 
 	if (!vop->is_enabled || !vop->lut || !vop->lut_regs)
 		return;
@@ -3731,7 +3731,7 @@ static int vop_win_init(struct vop *vop)
 			vop_area->nformats = vop_win->nformats;
 			vop_area->vop = vop;
 			vop_area->win_id = i;
-			vop_area->area_id = j;
+			vop_area->area_id = j + 1;
 			num_wins++;
 		}
 	}
-- 
2.35.3

